### DISCUSSION TITLE
@gw8674's Pool (MyAZPool)

### DISCUSSION THREAD

**Discussion by gw8674** [tagyoureit/nodejs-poolController#515] [My Pool]
I’ve been meaning to post my specs here for a while and finally found some time this weekend to get on with it.  

Just a plain and old (1987) rectangular pool with a PebbleTec & new tile facelift in 2007. Although, I’m pretty sure that I’m much prouder of what’s at the equipment pad and the electrons that make it all work rather than the pool/spa itself. <span style='font-size:40px;color:rgb(0,0,0);font-style:normal;font-weight:normal;'>&#9786;</span>

20K gallon 19’x36’ Cantilevered Play/Lap Pool & 6' dia detached IG spa. Caribbean Blue Baja Mini Pebble.
<img width="740" alt="2022-07-03_14-47-13" src="https://user-images.githubusercontent.com/52094163/177058570-565cf1f7-4b3a-461a-9e16-658452fcec62.png">
<img width="1193" alt="2022-07-03_14-54-59" src="https://user-images.githubusercontent.com/52094163/177058581-dbb46073-1f69-4a25-b90e-4ad308fa1863.png">

Pentair IntelliCenter i10PS
Pentair IntelliFlo VSF Filter Pump
Jandy MAX HP Spa Jet Pump
Pentair Clean & Clear 520 Filter
Pentair IntelliChlor IC-60
Pentair Chemical Acid Tank w/ Stenner Pump
(6) Pentair IntelliValves
Pentair MiniMax 100 Spa Heater
J&J Color Splash XG 120v LED’s
Maytronics Dolphin Active-60 Robotic Pool Cleaner
Solar Breeze Ariel Surface Robotic Cleaner
SP Auto Fill Leveler

**The important stuff**….
(1) Raspberry Pi 4B at the house running nodejs poolController and dashPanel.
(1) Raspberry Pi 4B in the enclosure running REM Chem Chemistry Controller. Stack includes: 
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(2) Whitebox Labs Tentacle T3’s
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(1) Sequent Microsystems Industrial I2C Automation HAT
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(1) Sequent Microsystems 4-Relay HAT
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;(1) Custom I/O board and I2C Extender/Active Terminator HAT
(1) Raspberry Pi 4B at the house running InfluxDB v1.8. Running Grafana locally and using VNC Viewer to access the RPi's.

Altelix 24x20x9 IP65 DIN Rail Fiberglass Enclosure w/ thermostatically controlled enclosure fans & 10” Touchscreen. 
Atlas Scientific Industrial pH, ORP & EC probes.
Atlas Scientific Filter Pressure Transducer, OAT & HUM sensors.
FlowVis Digital Flow Meter
Precision Digital ProVu Pulse Input Flow Rate/Totalizer connected to automatic pool fill/leveler
![0BBEEE7F-5438-4D38-BD30-4DD152E97E88_1_105_c](https://user-images.githubusercontent.com/52094163/177058757-830d8ce4-4bc3-4d92-9755-e2a8393baca1.jpeg)

![0100D6C3-76CB-4FC4-A872-09D06A27FA7A_1_105_c](https://user-images.githubusercontent.com/52094163/177058637-c18db341-e912-4bf4-83e0-667b9e7e07a5.jpeg)

![0616C88A-0FF2-47EA-AF32-43AA4328C9AB_1_105_c](https://user-images.githubusercontent.com/52094163/177058737-e8ce2aa2-8e94-4070-93f8-cf70f594ebad.jpeg)

<img width="1167" alt="2022-07-03_15-03-56" src="https://user-images.githubusercontent.com/52094163/177058672-89097709-5041-44e5-92d8-21bf05af43cf.png">

![C866425B-1302-484B-957D-0A47B65DB0D5_1_105_c](https://user-images.githubusercontent.com/52094163/177058794-c4b8ff58-8ff9-47b7-b04a-bf784ac99fb1.jpeg)

![6EAFB160-6AB6-4F9C-AB8C-86DEFF3AE5E4_1_105_c](https://user-images.githubusercontent.com/52094163/177060610-0819a616-51a8-4c7e-ac57-e2f1b6f30891.jpeg)

<img width="943" alt="2022-07-03_15-15-31" src="https://user-images.githubusercontent.com/52094163/177058999-a7a8ca8e-e93c-484d-aa60-a2f6920cd8db.png">

<img width="1734" alt="2022-07-03_15-16-22" src="https://user-images.githubusercontent.com/52094163/177059010-4b9eb4d1-68a3-4552-af34-06acb1d06d27.png">

Thanks very much to **rstrouse**, **tagyoureit** and a few others along the way for all of the tremendous help and guidance during the last few years.

**Comment by tagyoureit** [comment]
Love seeing the pics!  How do you have your plumbing setup such that you have the i10PS (shared body) with a single salt cell and filter but two separate bodies of water? It's a little hard to tell from the pics.

**Reply by gw8674** [reply]
@tagyoureit Thanks for all the help throughout!!
Good question. Actually it's only one body of water. The spa is connected to the pool via a 3" equalizer pipe installed between the pool and the spa. I have to put this rubber plug into that equalizer pipe before I use the spa, so the heated water does not flow out to the pool. It was the crazy way they did it around here back in the 80's I have learned.

The amount of water that is suctioned from and returned to each are controlled via the suction and return 3-way valves.

**Suction**
![2022-07-03_16-07-26](https://user-images.githubusercontent.com/52094163/177060271-13048f2c-5269-4d83-85d1-29f57067fdf2.jpg)


**Return**
![8CF9E300-062C-447D-B117-FD7C67AA659F_1_201_a](https://user-images.githubusercontent.com/52094163/177060194-ad300020-babf-4e06-b7d7-8b19d49f4ca3.jpeg)

**Comment by Telemachus1977** [comment]
That wiring is like a work of art.   No way I'm showing the inside of mine!

My old easytouch 4 (now an 8) has the breaker bus bars snack dab in the center.   And I don't use them.   I plan to remove them and put my din-bar there instead.

**Reply by Telemachus1977** [reply]
What application gives you all the analytical data/graphs?

I'm hoping the Pi Zero W that runs my system will have the processor power to do it.   I'm trying not to move up to my only spare Pi 4...

**Comment by gw8674** [comment]
@Telemachus1977
Thanks. But I can promise you that had I been wiring all that up in the middle of summer here in AZ, it probably wouldn't look the same. <span style='font-size:40px;color:rgb(0,0,0);font-style:normal;font-weight:normal;'>&#9786;</span>
Besides, as long as it all works, I guess it really doesn't matter what the wiring looks like. Yea, if you're not using the breaker bus bar, that definitely could be a good enclosure to install some goodies.

I'm using Grafana to present the data and graphs. It pulls the data from InfluxDB 1.8 which gets it from njspC and REM. I hope I got that right as I'm certainly no expert here. My knowledge regarding how it all works together is limited to the fact that "works great and last long time"  <span style='font-size:40px;color:rgb(0,0,0);font-style:normal;font-weight:normal;'>&#9786;</span>

Instead, I'll defer to @rstrouse and @tagyoureit who can really give you the technical explanations on how all it all ties together.

Not sure if your Pi Zero will work out in that respect. Again, I'll leave that to other's who have the technical prowess to provide you with some accurate answers.

**Reply by rstrouse** [reply]
Not sure I would run an Influx database on a Pi Zero.  But it does not matter where that resides anyway.  If you have a Server, NUC, Pi, PC, or Mac anywhere in the network (even the cloud version), you can point the njsPC instance to that Influx database wherever it may be and njsPC will batch update the time series automagically.   The same is true for Grafana or whatever visualization tool you use.

**Comment by Exit2Studios** [comment]
@gw8674 , why do you have insulation over your PH probe?

**Reply by tagyoureit** [reply]
The probes have "jelly" (technical term) in them that can freeze in cold temps, and could be subject to high temps as well.  @rstrouse mounted his on the back on his OCP and put a garbage bag over it (the power supply produces enough heat that it keeps the temps above 32F) so I copied him and have my covered, as well.  (I know that the probes can freeze from personal experiences.)

**Comment by gw8674** [comment]
@Exit2Studios 
Exactly what @tagyoureit said.
Also, my (water temp) EZO RTD Circuit obtains it's readings from the A.S. Industrial pH probe.
In my neck of the woods, I'm more concerned with high ambient temps (see current temp at my pad below <span style='font-size:40px;color:rgb(0,0,0);font-style:normal;font-weight:normal;'>&#9786;</span>), skewing the water temp readings being fed to the RTD since that temp value is a REM feed to pH and EC. 
![2022-07-11_17-53-01](https://user-images.githubusercontent.com/52094163/178385930-7e594911-edfc-48f2-9c9b-92e337688290.jpg)

But we have been known to drop down near freezing once in a blue moon, so definitely should help there as well.

**Reply by rstrouse** [reply]
Doing a bit better than you are today but even a wet fart comes out as dust at that humidity.
![image](https://user-images.githubusercontent.com/47839015/178387669-493eda0b-d5f9-4193-972e-e6cc86ef501d.png)

**Reply by gw8674** [reply]
For sure LOL...  and you win in all three categories! We're roasting over here!

**Reply by tagyoureit** [reply]
I thought a dry heat was better!

**Reply by rstrouse** [reply]
6% is pretty much a jerky dryer.

**Comment by spifftek70** [comment]
Could you comment on how you wired your filter gauge transducer in this configuration?

**Reply by spifftek70** [reply]
@gw8674 - I'm also in AZ.

**Reply by gw8674** [reply]
My Atlas Scientific EZO-PRS Pressure Sensor is connected to one of my two White Labs Tentacle T3] Carrier Boards that "stack" onto the Raspberry Pi.
The Pressure Sensor will need to be converted from its default UART protocol to I2C (address 106 (0x6A).
The instructions can be found in the EZO-PRS Data Sheet (https://files.atlas-scientific.com/EZO-PRS-Datasheet.pdf).
In addition, Atlas Scientific Tech Support (https://atlas-scientific.com/contact/) is very helpful in this regard as well.

I hope this is helpful to you.

Sorry for the delayed response. I don't check here much because my njsPc system is still running as good as it was 5 years ago. A testament to the developers!
